backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -157,7 +171,49 @@ def sanitize_records(self, records):
         properties2List(tests, ["issues", "platformsFailing", "archSummary"])
         properties2List(boots, ["issues", "platformsFailing", "archSummary"])
 
-        return (builds, tests, boots, trees)
+        return (builds, tests, boots)
+
+    def get_trees(self, hardware_id, origin, start_datetime, end_datetime):
+        tree_id_fields = [
+            "build__checkout__tree_name",
+            "build__checkout__git_repository_branch",
+            "build__checkout__git_repository_url",
+        ]
+
+        trees_query_set = Tests.objects.filter(
+            environment_compatible__contains=[hardware_id],
+            origin=origin,
+            start_time__gte=start_datetime,
+            start_time__lte=end_datetime,
```

Comment:
this filter should happen in the checkout table only
in case another batch of tests will be released for those trees
that way will mimic better the behavior in the tree monitoring